tests/buffer/buffer_api_common.h

```diff
@@ -436,12 +457,11 @@ void test_buffer(util::logger &log, cl::sycl::range<dims> &r,
 template <typename T, int numDims>
 void test_type_reinterpret(util::logger &log) {
   static constexpr size_t inputElemsPerDim = 4;
-  std::vector<uint8_t> reinterpretInputData(sizeof(T) * inputElemsPerDim *
-                                            numDims);
+  auto numElems = calc_elem_count<numDims>(inputElemsPerDim);
```

Comment:
... and just use:
```suggestion
  auto numElems = std::pow(inputElemsPerDim, dims);;
```

Reply:
`std::pow` uses one of floating point types, which doesn't seem to be a good fit here.

Reply:
Ah yes, I have interpreted the case (7) of https://en.cppreference.com/w/cpp/numeric/math/pow in a too optimistic way. :-(
But since the template stuff is not even used to produce a `constexpr` value, just use a loop?
```c++
auto numElems = inputElemsPerDim;
for (int i = 1; i < numDims; ++i)
  numElems *= inputElemsPerDim;
```

Reply:
Done.